pkg/operator/mon/mon.go

```diff
@@ -417,13 +536,31 @@ func (c *Cluster) loadMonConfig() error {
 	}
 
 	// Parse the max monitor id
-	if id, ok := cm.Data[maxMonIDKey]; ok {
+	if id, ok := cm.Data[MaxMonIDKey]; ok {
 		c.maxMonID, err = strconv.Atoi(id)
 		if err != nil {
 			logger.Errorf("invalid max mon id %s. %+v", id, err)
 		}
 	}
 
+	// Unmarshal mon mapping (node, port)
+	if mappingStr, ok := cm.Data[MappingKey]; ok {
+		if mappingStr != "" {
+			if err := json.Unmarshal([]byte(mappingStr), &c.mapping); err != nil {
+				logger.Errorf("invalid mapping json. json=%s; %+v", mappingStr, err)
+				c.mapping = &mapping{
+					Node: map[string]*nodeInfo{},
+					Port: map[string]int32{},
+				}
+			}
+		} else {
+			c.mapping = &mapping{
```

Comment:
will `mapping` be initialized if `ok` is false?

Reply:
Right, that is missing. Added it. Pushing the changes after I fixed all your comments.